### ISSUE TITLE
cabal-install 3.10, bundled with GHC 9.6, cannot be built with GHC 9.6

### ISSUE THREAD

**Issue by 9999years** [haskell/cabal#8872]
`cabal-install` 3.10 requires `base >=4.10 && <4.18`, but GHC 9.6 bundles `base-4.18` along with `cabal-install-3.10`. The version bounds should be bumped.

Note that the [GHC 9.6 release notes](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/9.6.1-notes.html) erroneously say that Cabal 3.9 is bundled with GHC 9.6; this is a mistake, as no Cabal 3.9 exists. The [GHC 9.6 download page](https://www.haskell.org/ghc/download_ghc_9_6_1.html) correctly notes that "GHC 9.6 only works with [cabal-install](http://hackage.haskell.org/package/cabal-install) version 3.10 or later." Although it doesn't entirely "work", as `cabal-install-3.10` cannot be built with GHC 9.6's `base` by default.

**Comment by andreabedini** [comment]
Thanks @9999years.

To be  pedantic, GHC 9.6 does not bundle `cabal-install-3.10.1.0`, it bundles `Cabal-3.10.1.0` and `Cabal-syntax-3.10.1.0`. Those packages have both a compatible version bound on base (>=4.9 && <5).

That said, yes, `cabal-install-3.10.1.0` and `cabal-install-solver-3.10.1.0` might have a overly restrictive bound on base.

The following seems to indicate that this is indeed the case
```
$ cabal install --installdir=$INSTALLDIR -w ghc-9.6 --allow-newer={cabal-install,cabal-install-solver}:base cabal-install-3.10.1.0
....
Completed    cabal-install-3.10.1.0 (exe:cabal)
Symlinking 'cabal' to '/tmp/tmp.edlG6NrwNO/cabal'
```

@Mikolaj could you relax the bound on base on Hackage for cabal-install and cabal-install-solver? (Perhaps after some testing, don't just take my word :P).

**Comment by Mikolaj** [comment]
Thank you for the report. We are trying to add GHC 9.6 to CI in https://github.com/haskell/cabal/pull/8840. The base bump is already there. I hope the PR will be ready before the cabal 3.10.2.0 release, but we have all kinds of technical problems.

I reported to GHC folk about the release notes flop.

**Comment by Mikolaj** [comment]
It was explained to me the canonical release notes location is https://downloads.haskell.org/ghc/9.6.1/docs/users_guide/9.6.1-notes.html. There the version is correct. The other location is transient and not up to date at this point.

**Comment by 9999years** [comment]
Can the 'transient' version be updated to indicate that it's non-canonical? As it stands, the 'transient' version is the first result for Googling "ghc 9.6 release notes".

**Comment by Mikolaj** [comment]
Indeed, it is so for me as well. That's unfortunate. Please file a GHC bug report.
